src/Kerbero.Common/Interactors/CreateNukiAccountInteractor.cs

```diff
@@ -0,0 +1,51 @@
+using Kerbero.Common.Entities;
+using Kerbero.Common.Exceptions;
+using Kerbero.Common.Models;
+using Kerbero.Common.Models.AccountMapper;
+using Kerbero.Common.Repositories;
+
+namespace Kerbero.Common.Interactors;
+
+/// <summary>
+/// This class provides a method to create a SmartLock nuki account, which must be saved to DB.
+/// </summary>
+public class CreateNukiAccountInteractor
+{
+	private readonly IAuthenticationRepository _repository;
+
+	public CreateNukiAccountInteractor(IAuthenticationRepository repository)
+	{
+		_repository = repository;
+	}
+
+	public async Task<int> Handle(NukiAccountDto nukiAccountDto)
+	{
+		var providerAccountEntity = NukiProviderAccountMapper.MapToEntity(nukiAccountDto);
+		if (!VerifyAccountInformationIntegrity(providerAccountEntity))
+		{
+			throw new InvalidTokenException();
+		}
+		_repository.AddSmartLockAccount(providerAccountEntity);
+		var writtenRows = await _repository.SaveChangesAsync();
+		if (writtenRows < 1)
+		{
+			throw new RepositoryIsNotBeenModified();
```

Comment:
```suggestion
			throw new CouldNotSaveNukiAccount(nukiAccount);
```
Do not let outside world know that we have a repository